src/helpers/string_helpers/mod.rs

```diff
@@ -0,0 +1,112 @@
+//! A Set of string utilities often required during code generation.
+//!
+//! See also: [`heck`](https://docs.rs/heck/latest/heck)
+//!
+//! `lowerCamelCase`: Convert a string to lowerCamelCase
+//! `upperCamelCase`: Convert a string to UpperCamelCase
+//! `snakeCase`: Convert a string to snake_case
+//! `kebabCase`: Conver a string to kebab-case
+//! `shoutySnakeCase`: Convert a string to SHOUTY_SNAKE_CASE
+//! `shoutyKebabCase`: Convert a string to SHOUTY-KEBAB-CASE
+//! `titleCase`: Convert a string to Title Case
+//! `trainCase`: Convert a string to Train-Case
+
+use heck::{
+    ToKebabCase, ToLowerCamelCase, ToShoutyKebabCase, ToShoutySnakeCase, ToSnakeCase, ToTitleCase,
+    ToTrainCase, ToUpperCamelCase,
+};
+
+macro_rules! define_case_helper {
+    ($helper_fn_name: ident, $heck_fn_name:ident) => {
+        pub(crate) fn $helper_fn_name(
+            h: &crate::render::Helper<'_>,
+            _: &crate::Handlebars<'_>,
+            _: &crate::context::Context,
+            _rc: &mut crate::render::RenderContext<'_, '_>,
+            out: &mut dyn crate::output::Output,
+        ) -> crate::helpers::HelperResult {
+            let param = h.param(0).and_then(|v| v.value().as_str()).unwrap_or("");
```

Comment:
Perhaps we can throw an error for non-string input here?

The error system is slightly different between 4.x and upcoming 5.x so I can do it for you.